src/yunohost/utils/password.py

```diff
@@ -0,0 +1,197 @@
+            else:
+                return ("error", "password_too_simple_" + str(self.validation_strength))
+
+        if self.strength < 3:
+            return ("warning", 'password_advice')
+        return ("success", "")
+
+    def compute(self, password, online=False):
+        # Indicators
+        length = len(password)
+        digits = 0
+        uppers = 0
+        lowers = 0
+        others = 0
+
+        for character in password:
+            if character in string.digits:
+                digits = digits + 1
+            elif character in ASCII_UPPERCASE:
+                uppers = uppers + 1
+            elif character in ASCII_LOWERCASE:
+                lowers = lowers + 1
+            else:
+                others = others + 1
+
+        # Check small list
+        unlisted = 0
+        if password not in SMALL_PWD_LIST:
+            unlisted = len(SMALL_PWD_LIST)
+
+        # Check big list
+        size_list = 100000
+        if unlisted > 0 and not self.is_in_cracklib_list(password, PWD_LIST_FILE):
+            unlisted = size_list if online else 320000000
+
+        # Check online big list
+        if unlisted > size_list and online and not self.is_in_online_pwned_list(password):
+            unlisted = 320000000
+
+        self.listed = unlisted < 320000000
+        return self.compare(unlisted, length, digits, lowers, uppers, others)
+
+    def compare(self, unlisted, length, digits, lowers, uppers, others):
+        strength = 0
+
+        for i, config in enumerate(self.strength_lvl):
+            if unlisted < config[0] or length < config[1] \
+               or digits < config[2] or lowers < config[3] \
+               or uppers < config[4] or others < config[5]:
+                break
+            strength = i + 1
+        return strength
+
+    def is_in_online_pwned_list(self, password, silent=True):
+        """
+        Check if a password is in the list of breached passwords from
+        haveibeenpwned.com
+        """
+
+        from hashlib import sha1
+        import requests
+        hash = sha1(password).hexdigest()
+        range = hash[:5]
+        needle = (hash[5:].upper())
+
+        try:
+            hash_list =requests.get('https://api.pwnedpasswords.com/range/' +
+                                      range, timeout=30)
+        except e:
+            if not silent:
+                raise
+        else:
+            if hash_list.find(needle) != -1:
+                return True
+        return False
+
+    def is_in_cracklib_list(self, password, pwd_dict):
+        try:
+            cracklib.VeryFascistCheck(password, None,
+                                      os.path.join(PWDDICT_PATH, pwd_dict))
+        except ValueError as e:
+            # We only want the dictionnary check of cracklib, not the is_simple
+            # test.
+            if str(e) not in ["is too simple", "is a palindrome"]:
+                return True
+
+
+class ProfilePasswordValidator(PasswordValidator):
```

Comment:
Eh are those three different classes really needed ? Isnt there any way to merge one or all of them together ? What are the differences between them ?